packages/shylu/shylu_dd/bddc/src/shylu_CoarseSpaceBDDC.h

```diff
@@ -64,15 +64,15 @@ using Teuchos::rcp;
 
 // Author: Clark R. Dohrmann
 namespace bddc {
-  
-template <class SX, class SM, class LO, class GO> 
+
+template <class SX, class SM, class LO, class GO>
   class CoarseSpaceBDDC
 {
 public:
   //
   // Convenience typedefs
   //
-  typedef Tpetra::DefaultPlatform::DefaultPlatformType::NodeType  Node;
+  typedef Tpetra::Map<>::node_type  Node;
```

Comment:
Is there a design document or a short paragraph the Tpetra team can point me that discusses the pros and cons of doing this ? I have two questions. 

(1) This sort of elevates the Map as the first object anyone should look at. The Map now serves dual purposes, what it originally did (how data is laid out in MPI) and it is sort of a handle for the Node type. Is this what is intended ? I worry about such dual uses, but I am fine with it if it makes the life of Tpeta team easier.
(2) Will there be support of execution space and memory space or just Node ?

Reply:
The eventual goal is to deprecate node and instead allow execution space and/or memory space to be provided.  This work is a partial step toward that goal.

Reply:
The following wiki page with Tpetra's deprecation plans may be helpful: https://github.com/trilinos/Trilinos/wiki/Deprecation1

Node is a leftover from Chris Baker's Kokkos 1.0 design, that persisted because interface changes are hard.  We plan to get rid of Node in favor of Kokkos execution space.  The Node instance never served any useful purpose in Tpetra other than managing Kokkos initialization and finalization, so now Map takes care of that.

Kokkos does not currently have full support for CUDA streams (e.g., `Kokkos::deep_copy` currently ignores its execution space instance argument).  Once this is fixed, Tpetra will consider letting users supply an execution space instance argument to computations.

Reply:
I wish the Map was not overloaded for this, but if this is temporary then it is fine. That link is very useful. I like the runtime choice. Hopefully, there will be a memory space enum as well.

Reply:
Map doesn't "overload" anything.  The only reason why it historically handled Kokkos initialization and finalization (and still does), is because the Teuchos unit test framework required it.